engine/client/client.go

```diff
@@ -681,13 +681,8 @@ func (c *Client) exportLogs(ctx context.Context, httpClient *httpClient) error {
 		if err := protojson.Unmarshal(data, &req); err != nil {
 			return fmt.Errorf("unmarshal spans: %w", err)
 		}
-
-		logs := telemetry.LogsFromPB(req.GetResourceLogs())
-
-		slog.ExtraDebug("received logs from engine", "len", len(logs))
-
-		if err := c.EngineLogs.Export(ctx, logs); err != nil {
-			return fmt.Errorf("export %d logs: %w", len(logs), err)
+		if err := enginetel.ReexportLogsFromPB(ctx, c.EngineLogs, &req); err != nil {
```

Comment:
Here our goal is to relay an OTLP logs payload from the receiving endpoint to another exporter, which expects a `[]sdklog.Record`. Previously we tried to construct that from the payload, but the only way to actually set the `Resource` on a `sdklog.Record` is to go through a `sdklog.LoggerProvider`. Fortunately in all cases that we want to do that we're about to send them off to a `log.Exporter` which is exactly what a `LoggerProvider` points to, so this `ReexportLogsFromPB` helper method does the conversion and exporting in one shot, and is used in both places.